### PR TITLE
Blog post on using ES6 and Vert.x

### PR DESCRIPTION
**Pull request by pmlopes**
This small post shows how to use the new JavaScript with Vert.x. It shows that our API is totally backward compatible and we can mix and match both ES5 and ES6 using a transpiler.

### CODE REVIEW

File: src/site/blog/posts/vertx3-es6-back-to-the-future.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+template: post.html
+date: 2015-11-23
+author: pmlopes
+draft: true
+---
+
+On [October 21th, 2015](http://backtothefuture.wikia.com/wiki/2015) we all rejoiced with the return from the past of
+Marty McFly with his flying car and so on, however in the Vert.x world we were quite sad that the JavaScript support we
+have was still using a technology released in [December 2009](https://en.wikipedia.org/wiki/ECMAScript#Versions). The
+support for ES5 is not something that we Vert.x team control but something that is inherited from running on top of
```
Comment (cescoffier) [inline src/site/blog/posts/vertx3-es6-back-to-the-future.md]: that the Vert.x team controls

File: src/site/blog/posts/vertx3-es6-back-to-the-future.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+[Nashorn](http://openjdk.java.net/projects/nashorn/).
+
+With all these nostalgic thoughts on my mind I've decided to bring us back to the future and by future I mean, lets
+start start using [EcmaScript6](http://es6-features.org/).
```
Comment (cescoffier) [inline src/site/blog/posts/vertx3-es6-back-to-the-future.md]: twice start

Comment (cescoffier) [inline src/site/blog/posts/vertx3-es6-back-to-the-future.md]: "lets start using a modern language for the browser" maybe

File: src/site/blog/posts/vertx3-es6-back-to-the-future.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+It turned out to be quite simple to achive this so I'll pick the hello world example and write it in ES6 just to show
```
Comment (cescoffier) [inline src/site/blog/posts/vertx3-es6-back-to-the-future.md]: achieve

File: src/site/blog/posts/vertx3-es6-back-to-the-future.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    "start": "./node_modules/.bin/vertx run lib/main.js"
+  },
+
+  "dependencies": {
+    "vertx3-full": "3.1.0",
+    "babel-cli": "6.2.0",
+    "babel-preset-es2015": "6.1.18"
+  }
+}
+```
```
Comment (cescoffier) [inline src/site/blog/posts/vertx3-es6-back-to-the-future.md]: As you can see, the main idea is to invoke the transpiler (Babel) when we are building our project, and run it using the generated files. This is slightly equivalent to a compilation process you would have using compiled language.

**Comment by pmlopes** [comment]
I am not sure if i should refer to typescript, i think it should deserve a post on its own.

**Comment by michel-kraemer** [comment]
@pmlopes Good idea. I'll see if I can write one on the weekend.

**Comment by michel-kraemer** [comment]
+1 for the updates

**Comment by cescoffier** [comment]
:+1:

**Comment by cescoffier** [comment]
let me know when you want it published.

**Comment by pmlopes** [comment]
We need to plan it so it does not go at the same time as your intro series,
we can either do it this week and next week your intro or vice versa


On Tue, Nov 24, 2015 at 5:16 PM, Clement Escoffier <notifications@github.com
> wrote:

> let me know when you want it published.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/vert-x3/vertx-web-site/pull/116#issuecomment-159318737>
> .
>



-- 
Paulo Lopes | Principal Software Engineer
Vert.x Developer
Red Hat

**Comment by cescoffier** [comment]
Let's publish yours today, and Mongo on Monday.

**Comment by cescoffier** [comment]
Here it is http://vertx.io/blog/vert-x-es6-back-to-the-future !

**Pull request merged**